Boundary Docs (#2907)
hashicorp/web-unified-docs: content/boundary/v1.0.x/content/docs/concepts/iam.mdx

---
layout: docs
page_title: Boundary identity and access management (IAM)
description: >-

>   Control access in Boundary with scopes, auth methods, accounts, users, groups, and roles — the six resources that make up Boundary's identity and access management (IAM) system, with example grant strings.

Reviewer: ```suggestion
  Learn how scopes, auth methods, accounts, users, groups, and roles make up Boundary's identity and access management (IAM) system, with example grant strings.
```

# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2025-11-18T17:04:41-05:00
last_modified: 2025-11-18T17:04:41-05:00
# END AUTO GENERATED METADATA
---

> # Boundary identity and access management (IAM)

Reviewer: ```suggestion
# Identity and access management (IAM)
```

Controlling who can access what gets unwieldy as teams, targets, and
environments grow. Boundary's identity and access management (IAM) system
solves this with six resources — scopes, auth methods, accounts, users,
groups, and roles — that together define authentication and the permissions

> granted after it.

Reviewer: ```suggestion
environments grow. Boundary's identity and access management (IAM) system
solves this with six resources that together define authentication and 
granted permissions: scopes, auth methods, accounts, users, groups, and roles.
```

It is recommended to add the user to a group, and then add the group to the role(s), instead of adding the user directly to the role(s). This way, you can manage multiple users at the same time, and it is easier to
change the permissions of the user by adding/removing them from groups. You can manage OIDC/LDAP users and managed groups the same way, directly in the provider.

## IAM examples by persona
Here are a few examples of access management for personas, using this sample scope structure. In this structure, there are 7 scopes overall: 1 global, 2 Orgs, and 4
Projects.

![Diagram showing a 7-scope hierarchy: 1 global scope containing 2 org scopes, each containing 2 project scopes.](/img/iam-example-structure.png)

> Each persona requires a role in the listed scope(s), with the user/group added as a principal and the listed grants applied:

Reviewer: ```suggestion
Each persona requires a role in the listed scopes, with the user or group added as a principal, and the listed grants applied:
```
